src/cursor/abstract_cursor.ts

```diff
@@ -220,6 +220,11 @@ export abstract class AbstractCursor<
     return this[kId] ?? undefined;
   }
 
+  /** @internal */
+  get isDead() {
```

Comment:
This makes much more sense to me to be a property of the cursor itself instead of passing a cursor to a function.

Reply:
Did we intentionally change the semantics of `isDead`?

Reply:
Nevermind, I see that now in the description.  But I'm still unsure why the logic was changed

Reply:
I changed it to exit the loop and enter clean-up logic at the correct times. The prior logic didn't account for a cursor already being closed, or killed, so the loop would attempt a getMore on a killed cursor id. This same "exit" condition applies to all the places the cursorIsDead helper was being used.